### PR TITLE
Prune caches to prevent bloat

### PR DESCRIPTION
**Pull request by Kotarski**
Calculate a usefulness score for each transform. If over a cache target size, prune entries starting with the lowest score until under the target size.

### CODE REVIEW

File: blockwork/build/caching.py [hunk 1]
```diff
@@ -293,6 +294,78 @@ def store_transform_to_any(ctx: Context, transform: "Transform", run_time: float
...
+        # as it is not useful to have it cached.
+        for transform, medials in transform_medials.items():
+            if medials - present_medials:
+                transform_scores[transform] = 0
+
+        # Remove transforms starting with those with the worst score until we
+        # reach the target size, always removing any with a score of zero.
+        for transform, score in sorted(transform_scores.items(), key=lambda i: i[1]):
+            if score > 0 and total_size < target_size:
+                # watermark_score = score
```
Comment (Intuity) [inline blockwork/build/caching.py:353]: Dead code?

File: blockwork/build/caching.py [hunk 2]
```diff
@@ -337,14 +410,22 @@ def drop_value(self, key: str) -> bool:
         return self.drop_item(key)
 
 
-    def fetch_value(self, key: str) -> Optional[str]:
+    def fetch_value(self, key: str, peek: bool=False) -> Optional[str]:
         '''
         Fetch a string value from the store by key. Return None if not present.
```
Comment (Intuity) [inline blockwork/build/caching.py:415]: Could really do with some markup of parameters and what they do

File: example/infra/caches.py [hunk 3]
```diff
@@ -14,6 +14,10 @@ def __init__(self, ctx: Context) -> None:
         self.cache_root.mkdir(exist_ok=True)
         self.content_store.mkdir(exist_ok=True)
 
+    @property
+    def target_size(self) -> int:
+        return 1024**3
```
Comment (Intuity) [inline example/infra/caches.py:19]: Is this worth adding as a parameter in the `site.yaml`?

**Review by Intuity — APPROVED** [review]

File: blockwork/build/caching.py (hunk 1 above)
Comment (Kotarski) [reply blockwork/build/caching.py]: Removed

File: blockwork/build/caching.py (hunk 2 above)
Comment (Kotarski) [reply blockwork/build/caching.py:421]: Added params to the cache docstrings

File: example/infra/caches.py (hunk 3 above)
Comment (Kotarski) [reply example/infra/caches.py:19]: Put it here for now as needed to go somewhere... but on the fence about whether it makes sense to go in the site.yaml as the target could be very different on a different machine/setup.

Added to issue #97

**Pull request merged**